### PR TITLE
Extract clique from network 

### PR DESCRIPTION
**Pull request by Yithis**
# Description

Move code related to creating network clique for validators out of `finality-aleph`.

## Type of change

- Refactor (non-breaking change)

# Checklist:

- I have made neccessary updates to the Infrastructure
- I have made corresponding changes to the existing documentation

### CODE REVIEW

File: finality-aleph/Cargo.toml [hunk 1]
```diff
@@ -14,6 +14,7 @@ current-aleph-bft-rmc = { package = "aleph-bft-rmc", version = "0.6" }
 legacy-aleph-bft = { package = "aleph-bft", version = "0.19" }
 legacy-aleph-bft-rmc = { package = "aleph-bft-rmc", version = "0.5" }
 
+aleph-clique = { package = "clique-network", path = "../clique" }
```
Comment (timorl) [inline finality-aleph/Cargo.toml:17]: I think importing it just as `clique-network` might be slightly better anyway, there is nothing _that_ Aleph-specific there.

File: clique/src/service.rs [hunk 2]
```diff
@@ -1,26 +1,23 @@
-use std::fmt::Debug;
+use std::{fmt::Debug, time::Duration};
 
 use futures::{
     channel::{mpsc, oneshot},
     StreamExt,
 };
 use log::{info, trace, warn};
+use sc_service::SpawnTaskHandle;
```
Comment (timorl) [inline clique/src/service.rs:8]: Ouch, cannot this be abstracted away with a trait?

**Review by timorl — APPROVED** [review]
Lots of small comments, but in general looks good enough to use.

File: clique/LICENSE [hunk 3]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
[the remark this answers is not in the document]
Comment (Yithis) [reply clique/LICENSE:1]: I thought so? Look at `aleph-node/primitives`

File: clique/README.md [hunk 4]
```diff
@@ -0,0 +1 @@
+A finality gadget using AlephBFT consensus mechanism from this [crate ](https://crates.io/crates/aleph-bft).
```
[the remark this answers is not in the document]
Comment (Yithis) [reply clique/README.md]: Fixed

File: clique/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+parking_lot = "0.12"
+rand = "0.8"
+serde = "1.0"
+tiny-bip39 = "1.0"
+tokio = { version = "1.25", features = [ "sync", "macros", "time", "rt-multi-thread", "io-util", "net" ] }
+
+sc-service = { git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.38" }
+
+[features]
+only_legacy = []
```
[the remark this answers is not in the document]
Comment (Yithis) [reply clique/Cargo.toml]: Fixed

File: clique/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+[package]
+name = "clique-network"
+version = "0.1.0"
+authors = ["Cardinal Cryptography"]
+edition = "2021"
+license = "Apache 2.0"
+
+[dependencies]
+# fixed version to 'freeze' some types used in abft, mainly `SignatureSet` used in justification and signature aggregation
```
[the remark this answers is not in the document]
Comment (Yithis) [reply clique/Cargo.toml]: Fixed

File: clique/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+/// Abstraction for requesting own network addressing information.
+pub trait NetworkIdentity {
+    type PeerId: PeerId;
+    type AddressingInformation: AddressingInformation<PeerId = Self::PeerId>;
+
+    /// The external identity of this node.
+    fn identity(&self) -> Self::AddressingInformation;
+}
+
+mod crypto;
```
[the remark this answers is not in the document]
Comment (Yithis) [reply clique/src/lib.rs]: Fixed

File: finality-aleph/Cargo.toml (hunk 1 above)
Comment (Yithis) [reply finality-aleph/Cargo.toml]: Fixed with `network-clique` name

**Comment by lesniak43** [comment]
maybe you should bump `finality-aleph` version (and all other, if necessary)?

**Comment by Yithis** [comment]
> maybe you should bump `finality-aleph` version (and all other, if necessary)?

Done, I don't think any other package is relevant

**Review by lesniak43 — APPROVED** [review]

**Pull request merged**